<launch>

  <arg name="display_output" default="true"/>
  <arg name="enable_gpu" default="false"/>
  <arg name="powersave" default= "0"/>
  <arg name="thread" default= "2"/>
  <arg name="target_size" default= "640"/>
  <arg name="camera_topic" default="/usb_cam/image_raw"/>

  <node name="yolox_node" pkg="ncnn_ros" type="yolox_node" output="screen">
    <param name="models_path" value="/home/bingda/ncnn-assets/models/"/>
    <param name="model_file" value="yolox.bin"/>
    <param name="param_file" value="yolox.param"/>
    <param name="display_output" value="$(arg display_output)"/>
    <param name="enable_gpu" value="$(arg enable_gpu)"/>
    <!-- powersave可以设置为0或者1或者2,其中0代表使用所有的cpu核,1代表只使用little cores,2代表只使用big cores-->
    <param name="powersave" value="$(arg powersave)"/>    
    <param name="thread" value="$(arg thread)"/>   
    <param name="target_size" value="640"/>
    <param name="prob_threshold" value="0.35"/>
    <remap from="/camera/image_raw" to="$(arg camera_topic)"/>
  </node>

</launch>
